File: examples/tracing/dotnet/example/BikeService.cs
namespace Example;

internal class BikeService
{
    private readonly OrderService _orderService;

    public BikeService(OrderService orderService)
    {
        _orderService = orderService;
    }

    public void Order(int searchRadius)
    {
        _orderService.FindNearestVehicle(searchRadius, "bike");
    }
}

File: examples/tracing/java-wall/src/main/java/org/example/rideshare/bike/BikeService.java
package org.example.rideshare.bike;

import org.example.rideshare.OrderService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class BikeService {
    @Autowired
    OrderService orderService;

    public void orderBike(int searchRadius) {
        orderService.findNearestVehicle(searchRadius, "bike");
    }
}
